pkg/codegen/internal/test/testdata/output-funcs/go/codegen_test.go

```diff
@@ -0,0 +1,256 @@
+					expand),
+			},
+		}
+		result := ListStorageAccountKeysResult{
+			Keys: inputs,
+		}
+		outputs := map[string]interface{}{
+			"keys": result.Keys,
+		}
+		return resource.NewPropertyMapFromMap(outputs), nil
+	}
+
+	if args.Token == "madeup-package:codegentest:funcWithDefaultValue" ||
+		args.Token == "madeup-package:codegentest:funcWithAllOptionalInputs" ||
+		args.Token == "madeup-package:codegentest:funcWithListParam" ||
+		args.Token == "madeup-package:codegentest:funcWithDictParam" {
+		result := FuncWithDefaultValueResult{
+			R: fmt.Sprintf("%v", args.Args),
+		}
+		outputs := map[string]interface{}{
+			"r": result.R,
+		}
+		return resource.NewPropertyMapFromMap(outputs), nil
+	}
+
+	if args.Token == "azure-native:codegentest:getIntegrationRuntimeObjectMetadatum" {
+		targs := GetIntegrationRuntimeObjectMetadatumArgs{}
+		for k, v := range args.Args {
+			switch k {
+			case "factoryName":
+				targs.FactoryName = v.V.(string)
+			case "integrationRuntimeName":
+				targs.IntegrationRuntimeName = v.V.(string)
+			case "metadataPath":
+				metadataPath := v.V.(string)
+				targs.MetadataPath = &metadataPath
+			case "resourceGroupName":
+				targs.ResourceGroupName = v.V.(string)
+			}
+		}
+		nextLink := "my-next-link"
+		result := GetIntegrationRuntimeObjectMetadatumResult{
+			NextLink: &nextLink,
+			Value:    []interface{}{targs},
+		}
+		outputs := map[string]interface{}{
+			"nextLink": result.NextLink,
+			"value":    []interface{}{fmt.Sprintf("factoryName=%s", targs.FactoryName)},
+		}
+
+		return resource.NewPropertyMapFromMap(outputs), nil
+	}
+
+	panic(fmt.Errorf("Unknown token: %s", args.Token))
+}
+
+func TestListStorageAccountKeysOutput(t *testing.T) {
+	pulumiTest(t, func(ctx *pulumi.Context) error {
+		output := ListStorageAccountKeysOutput(ctx, ListStorageAccountKeysOutputArgs{
+			AccountName:       pulumi.String("my-account-name"),
+			ResourceGroupName: pulumi.String("my-resource-group-name"),
+		})
+
+		keys := waitOut(t, output.Keys()).([]StorageAccountKeyResponse)
+
+		assert.Equal(t, 1, len(keys))
+		assert.Equal(t, "key", keys[0].KeyName)
+		assert.Equal(t, "permissions", keys[0].Permissions)
+		assert.Equal(t, "accountName=my-account-name, resourceGroupName=my-resource-group-name, expand=",
+			keys[0].Value)
+
+		output = ListStorageAccountKeysOutput(ctx, ListStorageAccountKeysOutputArgs{
+			AccountName:       pulumi.String("my-account-name"),
+			ResourceGroupName: pulumi.String("my-resource-group-name"),
+			Expand:            pulumi.String("my-expand"),
+		})
+
+		keys = waitOut(t, output.Keys()).([]StorageAccountKeyResponse)
+
+		assert.Equal(t, 1, len(keys))
+		assert.Equal(t, "key", keys[0].KeyName)
+		assert.Equal(t, "permissions", keys[0].Permissions)
+		assert.Equal(t, "accountName=my-account-name, resourceGroupName=my-resource-group-name, expand=my-expand",
+			keys[0].Value)
+
+		return nil
+	})
+}
+
+// TODO: it seems that default values are not supported by Go codegen
```

Comment:
Nit: We tend to format these like:

```go
// TODO[pulumi/pulumi#7811]: 
```

Reply:
Will do, very curious is there a tool that consumes these?